app/controllers/api/v1/users_controller.rb

```diff
@@ -29,6 +29,48 @@ def update
         end
       end
 
+      # For social media logins, renders the appropriate `redirect_to` path depending on whether the user is registered.
+      #
+      # Requires that the ActionController::Parameters contain a :user key, with a nested :email key.
+      # For example: { user: { email: "john@example.com" } }
+      #
+      # @return [String] A string of the user's redirect_to path
+      # @see https://github.com/zquestz/omniauth-google-oauth2#devise
+      #
+      def exist
+        user = User.find_by(email: params[:user][:email])
```

Comment:
Cool method on hash is [dig](http://ruby-doc.org/core-2.3.0_preview1/Hash.html#method-i-dig).

This will allow you to fetch a key that is nested in a hash, while simultaneously protecting against any upstream keys being `nil`.

For example, if for some reason `params[:user]` was `nil`, this would blow up.

But with `dig`, you can call:

```ruby
params.dig(:user, :email)
```

And if `:user` is not present, the call would return `nil`, and not blow up.  It is kind of like `:try`, for hashes.

Reply:
done, and changed to this pattern for the other key fetches.